src/spikeinterface/postprocessing/template_metrics.py

```
@@ -116,8 +116,6 @@ def _set_params(
         **other_kwargs,
     ):
 
-        import pandas as pd
```

The import should throw an error at the init of anything that will need it downstream. Is this the case, here?

---

If that's the case we can check check for the spec and raise an error. I can add that in a new commit.

---

I am just wondering if this is a place where some things are initialized but I don't know template metrics.

---

We are currently using the package at all. I think because we use it later Sam/Alessio just import it everywhere. Tests are passing so we are not directly using it at init. But you raise a good point that we should probably check that it is around at init for users!